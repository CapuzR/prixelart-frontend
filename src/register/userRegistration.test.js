import React from 'react';
import UserRegistration from "./userRegistration";
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import axiosMock from "axios";
jest.mock("axios");

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

beforeEach(()=> {
  render(<UserRegistration />);
});

// it('<UserRegistration /> renders correctly', () => {
//   const tree = renderer
//     .create(<UserRegistration />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

describe('with invalid input values', () => {

  it('display required errors for username, email, firstName, lastName, password and T&C when empty', ()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);
    const register = screen.getByText(/Registrarme/i);

    expect(username.textContent).toBe("");
    expect(firstName.textContent).toBe("");
    expect(lastName.textContent).toBe("");
    expect(email.textContent).toBe("");
    expect(password.textContent).toBe("");
    
    fireEvent.change(username, { target: { value: '' } });
    fireEvent.change(firstName, { target: { value: '' } });
    fireEvent.change(lastName, { target: { value: '' } });
    fireEvent.change(email, { target: { value: '' } });
    fireEvent.change(password, { target: { value: '' } });
    fireEvent.submit(register);

    expect(screen.getByText('Por favor completa todos los campos requeridos.')).toBeTruthy();
  });

  it('invalid email: no @ nor .domain', async ()=> {  
    const emailField = screen.getByLabelText(/Correo/i);
    expect(emailField.textContent).toBe("");
    fireEvent.change(emailField, { target: { value: 'Test', textContent: 'Test' } });
    expect(screen.getByText('Por favor introduce un correo electrónico válido.')).toBeTruthy();
  });

  it('invalid email: no .domain', async ()=> {  
    const emailField = screen.getByLabelText(/Correo/i);
    expect(emailField.textContent).toBe("");
    fireEvent.change(emailField, { target: { value: 'Test@fkjsd', textContent: 'Test@fkjsd' } });
    expect(screen.getByText('Por favor introduce un correo electrónico válido.')).toBeTruthy();
  });

  it('invalid email: no @', async ()=> {  
    const emailField = screen.getByLabelText(/Correo/i);
    expect(emailField.textContent).toBe("");
    fireEvent.change(emailField, { target: { value: 'Test.com', textContent: 'Test.com' } });
    expect(screen.getByText('Por favor introduce un correo electrónico válido.')).toBeTruthy();
  });

  it('no characters after punctuation', async ()=> { 
    const emailField = screen.getByLabelText(/Correo/i); 
    expect(emailField.textContent).toBe("");
    fireEvent.change(emailField, { target: { value: 'Test@gmail.', textContent: 'Test@gmail.' } });
    expect(screen.getByText('Por favor introduce un correo electrónico válido.')).toBeTruthy();
  });

  it('invalid password',async ()=> {
    const passwordField = screen.getByLabelText("Contraseña");

    expect(passwordField.textContent).toBe("");
    
    fireEvent.change(passwordField, { target: { value: 'Test' } });


    //Más adelante sería ideal ubicar por palabras claves y no por textos completos para poder modificar detalles.
    expect(screen.getByText('Disculpa, tu contraseña debe tener entre 8 y 15 caracteres, incluyendo al menos: una minúscula, una mayúscula, un número y un caracter especial.')).toBeTruthy();
  });
});

describe('with valid input values', () => {
  it('Inputting form login input states works correctly', ()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);

    fireEvent.change(username, {target: {value: 'rcapuz', textContent: 'rcapuz'}});
    fireEvent.change(firstName, {target: {value: 'Ricardo', textContent: 'Ricardo'}});
    fireEvent.change(lastName, {target: {value: 'Capuz', textContent: 'Capuz'}});
    fireEvent.change(email, {target: {value: 'rcapuz@prixelart.com', textContent: 'rcapuz@prixelart.com'}});
    fireEvent.change(password, {target: {value: 'Test123.*', textContent: 'Test123.*'}});

    expect(username.textContent).toBe('rcapuz');
    expect(firstName.textContent).toBe('Ricardo');
    expect(lastName.textContent).toBe('Capuz');
    expect(email.textContent).toBe('rcapuz@prixelart.com');
    expect(password.textContent).toBe('Test123.*');
  });
  
  it('calls api with data from form with all fields', async()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);
    const register = screen.getByText(/Registrarme/i);
    const axMock = axiosMock.post.mockResolvedValue({data: {success: true}});

    expect(username.textContent).toBe('');
    expect(firstName.textContent).toBe('');
    expect(lastName.textContent).toBe('');
    expect(email.textContent).toBe('');
    expect(password.textContent).toBe('');

    await waitFor(async()=> {
      fireEvent.change(username, {target: {value: 'rcapuz', textContent: 'rcapuz'}});
      fireEvent.change(firstName, {target: {value: 'Ricardo', textContent: 'Ricardo'}});
      fireEvent.change(lastName, {target: {value: 'Capuz', textContent: 'Capuz'}});
      fireEvent.change(email, {target: {value: 'rcapuz@prixelart.com', textContent: 'rcapuz@prixelart.com'}});
      fireEvent.change(password, {target: {value: 'Test123.*', textContent: 'Test123.*'}});
    });
    
    await waitFor(async()=> {
      fireEvent.click(register);
    });

    expect(axMock).toHaveBeenCalled();
  });
  
  it('username already registered', async()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);
    const register = screen.getByText(/Registrarme/i);
    axiosMock.post.mockResolvedValue({data: {info: 'error_username', message: 'Disculpa, el nombre de usuario ya está registrado.'}})

    expect(username.textContent).toBe('');
    expect(firstName.textContent).toBe('');
    expect(lastName.textContent).toBe('');
    expect(email.textContent).toBe('');
    expect(password.textContent).toBe('');

    await waitFor(async()=> {
      fireEvent.change(username, {target: {value: 'rcapuz', textContent: 'rcapuz'}});
      fireEvent.change(firstName, {target: {value: 'Ricardo', textContent: 'Ricardo'}});
      fireEvent.change(lastName, {target: {value: 'Capuz', textContent: 'Capuz'}});
      fireEvent.change(email, {target: {value: 'rcapuz@prixelart.com', textContent: 'rcapuz@prixelart.com'}});
      fireEvent.change(password, {target: {value: 'Test123.*', textContent: 'Test123.*'}});
    });
    
    await waitFor(async()=> {
      fireEvent.click(register);
    });

    expect(screen.getByText('Disculpa, el nombre de usuario ya está registrado.')).toBeTruthy();
  });

  it('email already registered', async()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);
    const register = screen.getByText(/Registrarme/i);
    axiosMock.post.mockResolvedValue({data: {info: 'error_email', message: 'Disculpa, el correo del usuario ya está registrado.'}})

    expect(username.textContent).toBe('');
    expect(firstName.textContent).toBe('');
    expect(lastName.textContent).toBe('');
    expect(email.textContent).toBe('');
    expect(password.textContent).toBe('');

    await waitFor(async()=> {
      fireEvent.change(username, {target: {value: 'rcapuz', textContent: 'rcapuz'}});
      fireEvent.change(firstName, {target: {value: 'Ricardo', textContent: 'Ricardo'}});
      fireEvent.change(lastName, {target: {value: 'Capuz', textContent: 'Capuz'}});
      fireEvent.change(email, {target: {value: 'rcapuz@prixelart.com', textContent: 'rcapuz@prixelart.com'}});
      fireEvent.change(password, {target: {value: 'Test123.*', textContent: 'Test123.*'}});
    });
    
    await waitFor(async()=> {
      fireEvent.click(register);
    });

    expect(screen.getByText('Disculpa, el correo del usuario ya está registrado.')).toBeTruthy();
  });
  
  it('redirects to prixer Registration', async ()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);
    const register = screen.getByText(/Registrarme/i);
    const pageTitle = screen.getByText("Registrar");
    axiosMock.post.mockResolvedValue({data: {success: true}});

    expect(pageTitle.textContent).toBe("Registrar");

    await waitFor(async()=> {
      fireEvent.change(username, {target: {value: 'rcapuz', textContent: 'rcapuz'}});
      fireEvent.change(firstName, {target: {value: 'Ricardo', textContent: 'Ricardo'}});
      fireEvent.change(lastName, {target: {value: 'Capuz', textContent: 'Capuz'}});
      fireEvent.change(email, {target: {value: 'rcapuz@prixelart.com', textContent: 'rcapuz@prixelart.com'}});
      fireEvent.change(password, {target: {value: 'Test123.*', textContent: 'Test123.*'}});
    });
    
    await waitFor(async()=> {
      fireEvent.click(register);
    });

    expect(mockHistoryPush).toHaveBeenCalledWith({pathname: '/registrar/prixer'});
  });
  
  it('renders snackbar with success message', async()=> {
    const username = screen.getByLabelText(/Usuario/i);
    const firstName = screen.getByLabelText(/Nombre/i);
    const lastName = screen.getByLabelText(/Apellido/i);
    const email = screen.getByLabelText(/Correo electrónico/i);
    const password = screen.getByLabelText(/Contraseña/i);
    const register = screen.getByText(/Registrarme/i);
    axiosMock.post.mockResolvedValue({data: {success: false}})
  
    expect(username.textContent).toBe('');
    expect(firstName.textContent).toBe('');
    expect(lastName.textContent).toBe('');
    expect(email.textContent).toBe('');
    expect(password.textContent).toBe('');

    await waitFor(async()=> {
      fireEvent.change(username, {target: {value: 'rcapuz', textContent: 'rcapuz'}});
      fireEvent.change(firstName, {target: {value: 'Ricardo', textContent: 'Ricardo'}});
      fireEvent.change(lastName, {target: {value: 'Capuz', textContent: 'Capuz'}});
      fireEvent.change(email, {target: {value: 'rcapuz@prixelart.com', textContent: 'rcapuz@prixelart.com'}});
      fireEvent.change(password, {target: {value: 'Test123.*', textContent: 'Test123.*'}});
    });

    await waitFor(async()=> {
      fireEvent.click(register);
    });


    expect(screen.queryByText('Registro de usuario exitoso.')).toBeInTheDocument();
  });
});