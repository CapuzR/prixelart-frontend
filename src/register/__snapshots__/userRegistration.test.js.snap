// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRegistration /> renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Registrar
    </h1>
    <form
      className="makeStyles-form-3"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="username"
              id="username-label"
            >
              Usuario
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="username"
                name="username"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Usuario
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="firstName"
              id="firstName-label"
            >
              Nombre
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="fname"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="firstName"
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Nombre
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="lastName"
              id="lastName-label"
            >
              Apellido
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="lname"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="lastName"
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Apellido
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Correo electrónico
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="email"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Correo electrónico
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
            xs={12}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="outlined-adornment-password"
            >
              Contraseña
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                id="outlined-adornment-password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Contraseña
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <label
            className="MuiFormControlLabel-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  className="PrivateSwitchBase-input-13"
                  data-indeterminate={false}
                  onChange={[Function]}
                  type="checkbox"
                  value="allowExtraEmails"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Acepto las condiciones de uso.
            </span>
          </label>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
        value="submit"
      >
        <span
          className="MuiButton-label"
        >
          Registrar
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            ¿Ya tienes una cuenta? Inicia sesión.
          </a>
        </div>
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root MuiBox-root-14"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://prixelart.com/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        prixelart.com
      </a>
       
      2020
      .
    </p>
  </div>
</main>
`;
